pennylane/pauli/conversion.py

```diff
@@ -346,6 +346,17 @@ def pauli_sentence(op):
     return _pauli_sentence(op)
 
 
+def is_pauli_sentence(op):
+    """Returns True of the operator is a PauliSentence and False otherwise."""
+    if op._pauli_rep is not None:  # pylint: disable=protected-access
+        return True
+    if not isinstance(op, (Hamiltonian, Sum)):
+        return False
+    if isinstance(op, Hamiltonian):
+        return all(is_pauli_word(o) for o in op.ops)
+    return all(is_pauli_word(o) for o in op)
```

Comment:
this is pre-computed on `Sum` objects on initialization, so you don't need to validate them. if a `Sum` is a pauli sentence, its `_pauli_rep` will be defined, so you'll never make it past the first condition

Reply:
Understood, will fix that.